### PR TITLE
Static check build fixes

### PR DESCRIPTION
**Pull request by wallyqs**
Addresses the nits from staticcheck which show up in the build

### CODE REVIEW

File: test/conn_test.go [hunk 1]
```diff
@@ -375,7 +375,7 @@ func TestErrOnConnectAndDeadlock(t *testing.T) {
 		nc, err := nats.Connect(natsURL)
 		if err == nil {
 			nc.Close()
-			t.Fatal("Expected bad INFO err, got none")
```
Comment (derekcollison) [inline test/conn_test.go]: Again I think we want fatal here.

File: nats.go [hunk 2]
```diff
@@ -1445,7 +1445,7 @@ func (nc *Conn) waitForMsgs(s *Subscription) {
 		}
 
 		// Deliver the message.
-		if m != nil && (max <= 0 || delivered <= max) {
```
Comment (derekcollison) [inline nats.go]: What if max < 0 ?

Comment (wallyqs) [inline nats.go]: This addresses following staticcheck report `github.com/nats-io/nats/nats.go:1448:19: 'x <= 0' for unsigned values of x is the same as 'x == 0'`. Adding a `max < 0` reports that `unsigned values are never < 0` too...

**Comment by wallyqs** [comment]
Closing as needs to address first with proper fix for usage of T.Fatalf in goroutine error (`the goroutine calls T.Fatalf, which must be called in the same goroutine as the test` issue)

File: nats.go (hunk 2 above)
Comment (derekcollison) [inline nats.go]: Gotcha..

**Pull request closed without merging**